### PR TITLE
New topic flow - FE & BE integration

### PR DESCRIPTION
**Pull request by masvelio**
## Link to Issue
Closes: #9224 

## Description of Changes
- wired up frontend flow and forms to backend mutations and queries

## Test Plan
- the whole flow described [here ](https://www.figma.com/board/xsXNa51D2oFNjOpXr5AveR/Weighted-Voting-topic-creation-flow?node-id=0-1&node-type=canvas&t=IwlxCSsft9czGgyd-0)should be testable now
- enable flag FLAG_FARCASTER_CONTEST=true
- create community (you should not be able to enable stake in create community flow anymore)

### 1. Create regular topic
- go to `topics` menu
- fill out the form and click `next`
- click `no, create regular topic`
- you should be redirected to the topic page

https://github.com/user-attachments/assets/85ac22f7-5a92-4128-8fe2-120aebc72037

### 2. Create erc20 topic
- go to `topics` menu
- fill out the form and click `next`
- click `yes` to enable weighted voting
- select `connect ERC20 token` button and click `continue`
- fill out ERC20 form (eg use this address 0x19848077f45356b21164c412eff3d3e4ff6ebc31)
- click `enable weighted voting for topic` 
- you should be redirected to the topic page and you should see the banner on the top of the page with token information

https://github.com/user-attachments/assets/7a289e9d-aab8-4dad-9595-0c705aa3c55d

### 3. Create stake topic if stake is not enabled
- go to `topics` menu
- fill out the form and click `next`
- click `yes` to enable weighted voting
- select `use community stake` button and click `continue`
- enable stake by signing two transactions
- on success you should see the page with stake details 
- click `Create topic with stakes` 
- you should be redirected to the topic page

https://github.com/user-attachments/assets/2f3f2188-5298-4de6-999a-1dfb20f14228

### 4. Create stake topic if stake is enabled
- this is the same as numer 3 ☝️ but you don't need to sign transactions

## Deployment Plan

1. n/a

## Other Considerations

- n/a

### CODE REVIEW

File: packages/commonwealth/client/scripts/state/api/topics/createTopic.ts [hunk 1]
```diff
@@ -3,11 +3,13 @@ import axios from 'axios';
 import Topic from 'models/Topic';
 import app from 'state';
 import { ApiEndpoints, SERVER_URL, queryClient } from 'state/api/config';
+import { TopicForm } from 'views/pages/CommunityManagement/Topics/Topics';
 import { userStore } from '../../ui/user';
 
-interface CreateTopicProps {
+interface CreateTopicProps extends TopicForm {
+  // TODO remove those types below when the CreateTopicSectionOld is removed
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/state/api/topics/createTopic.ts:10]: lets add a cleanup ticket reference here

**Review by mzparacha — APPROVED** [review]

**Review by Israellund — APPROVED** [review]

File: packages/commonwealth/client/scripts/state/api/topics/createTopic.ts (hunk 1 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/state/api/topics/createTopic.ts]: actually after resolving conflicts, this is not a case anymore as we switched to trpc

File: packages/commonwealth/client/scripts/views/pages/CommunityManagement/Topics/Topics.tsx [hunk 2]
```diff
@@ -10,35 +11,111 @@ import WVERC20Details from './WVERC20Details';
...
+        defaultOffchainTemplate: '',
+        isPWA: isAddedToHomeScreen,
+        ...erc20,
+        ...stake,
+      });
+
+      navigate(`/discussions/${encodeURI(topicFormData.name.trim())}`);
+    } catch (err) {
+      notifyError('Failed to create topic');
+      console.error(err);
```
[the remark this answers is not in the document]
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/CommunityManagement/Topics/Topics.tsx:94]: we dont have logger on the front end for now, this is something we definitely can take a look at

File: packages/commonwealth/client/scripts/views/pages/CommunityManagement/StakeIntegration/StakeIntegration.tsx [hunk 3]
```diff
@@ -60,9 +64,14 @@ const StakeIntegration = ({
...
-    console.log('create topic');
-    // TODO temp solution - will be integrated in upcoming PR
-    navigate('/discussions');
+    onCreateTopic({
+      stake: {
+        weightedVoting: TopicWeightedVoting.Stake,
+      },
+    }).catch((err) => {
+      notifyError('Failed to create topic');
+      console.log(err);
```
[the remark this answers is not in the document]
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/CommunityManagement/StakeIntegration/StakeIntegration.tsx:73]: we dont have logger on the front end for now, this is something we definitely can take a look at

File: packages/commonwealth/client/scripts/state/api/communities/updateCommunity.ts [hunk 4]
```diff
@@ -93,6 +96,33 @@ type UseUpdateCommunityMutationProps = {
   reInitAppOnSuccess?: boolean;
 };
 
+export const DEPRECATED_useUpdateCommunityMutation = ({
```
[the remark this answers is not in the document]
Comment (masvelio) [reply packages/commonwealth/client/scripts/state/api/communities/updateCommunity.ts]: thanks for this advice. Actually this commit was reverted so its not valid anymore

**Pull request merged**